inginious/frontend/pages/course_admin/api_tokens.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+from collections import OrderedDict
+
+import flask
+
+import jwt
+from datetime import datetime
+
+from inginious.frontend.pages.course_admin.utils import INGIniousAdminPage
+
+class CourseAPITokens(INGIniousAdminPage):
+    """ List information about api tokens """
+
+    def GET_AUTH(self, courseid):  # pylint: disable=arguments-differ
+        """ GET request """
+        course, __ = self.get_course_and_check_rights(courseid)
+        return self.page(course)
+
+    def POST_AUTH(self, courseid):
+        """ POST request """
+        course, __ = self.get_course_and_check_rights(courseid)
+        user_input = flask.request.form.copy()
+        username = self.user_manager.session_username()
+
+        if "add_token" in user_input:
+            descr = user_input.get("descr", "")
+            expire = user_input.get("expiration", "")
+            if expire == "":
+                expire = "no"
+            insert_id = self.database.tokens.insert_one({"courseid": courseid, "expire": str(expire), "description": descr, "username": username}).inserted_id
+            msg = str(self.GET_TOKEN(str(insert_id), courseid, username, expire))
+
+        else:
+            tok_id = user_input.get("token_id", "")
+            self.database.tokens.delete_one({"$expr": {"$eq": ["$_id", {"$toObjectId": tok_id}]}, "username": username})
+            msg = "removed"
+
+        return self.page(course, msg)
+
+    def GET_TOKEN(self, insert_id, courseid, username, expire):
```

Comment:
The `GET*` and `POST*` methods are the remnants of the `web.py` framework era where methods had to match HTTP methods. This does not respect PEP8 at all. Do not use capital letters. Morever, I'd rather call this one `generate` or `encode` rather than `get`